### PR TITLE
New billing cloud

### PR DESCRIPTION
**Pull request by zugao**
## Summary

This PR changes the way we bill our cloud services.
Instead of writing our billing metrics to a billing database, we now export the billing metrics directly into our Odoo.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by Kidswiss — COMMENTED** [review]
Some things I found while looking at this

File: pkg/cloudscale/metrics.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	trafficOutValue, err := convertUnit(units[productIdTrafficOut], uint64(bucketMetricsData.TimeSeries[0].Usage.SentBytes))
+	if err != nil {
+		return nil, err
+	}
+	queryRequestsValue, err := convertUnit(units[productIdQueryRequests], uint64(bucketMetricsData.TimeSeries[0].Usage.SentBytes))
+	if err != nil {
+		return nil, err
+	}
+
+	//TODO where to put zone and namespace?
```
[the remark this answers is not in the document]
Comment (zugao) [reply pkg/cloudscale/metrics.go:110]: As discussed already, will include the zone and the namespace in the group item field

**Review by Kidswiss — APPROVED** [review]
LGTM, just a few comments that still say billing database and some ideas for the future

File: pkg/control/control.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func GetSalesOrder(ctx context.Context, k8sClient client.Client, orgId string) (string, error) {
+
+	org := &orgv1.Organization{}
+	err := k8sClient.Get(ctx, client.ObjectKey{Name: orgId}, org)
+	if err != nil {
+		return "", fmt.Errorf("cannot get Organization object '%s', err: %v", orgId, err)
+	}
+	if org.Status.SaleOrderName != "" {
```
Comment (Kidswiss) [inline pkg/control/control.go:18]: IMHO this makes is a bit easier to read and you don't actually need an else branch.

This way the happy path ends with the function.

```suggestion
	if org.Status.SaleOrderName == "" {
    return "", fmt.Errorf("Cannot get SalesOrder from organization object '%s', err: %v", orgId, err)
  }
  return org.Status.SaleOrderName, nil
```

File: pkg/control/control.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package control
```
Comment (Kidswiss) [inline pkg/control/control.go:1]: Maybe call it `controlAPI`?

File: pkg/kubernetes/client.go [hunk 4]
```diff
@@ -49,3 +75,31 @@ func restConfig(kubeconfig string, url string, token string) (*rest.Config, erro
 	}
 	return &rest.Config{Host: url, BearerToken: token}, nil
 }
+
+func FetchNamespaceWithOrganizationMap(ctx context.Context, k8sClient client.Client) (map[string]string, error) {
```
Comment (Kidswiss) [inline pkg/kubernetes/client.go:79]: I'd probably move this to the `controlAPI` package, it doesn't really fit here.

**Review by Kidswiss — CHANGES_REQUESTED** [review]

File: pkg/kubernetes/client.go (hunk 4 above)
Comment (TheBigLee) [reply pkg/kubernetes/client.go:79]: Doesn't make much sense in the controlAPI package either, as it's not about Namespaces on the controlAPI

File: pkg/control/control.go (hunk 3 above)
Comment (TheBigLee) [reply pkg/control/control.go:1]: Fixed

File: pkg/kubernetes/client.go (hunk 4 above)
Comment (Kidswiss) [reply pkg/kubernetes/client.go:79]: Yeah it doesn't really fit anywhere I think... ok, let's leave it here.

**Review by Kidswiss — APPROVED** [review]
Please fixup your `f` commits.

The rest LGTM.

File: pkg/exoscale/dbaas.go [hunk 5]
```diff
@@ -225,60 +191,62 @@ func aggregateDBaaS(ctx context.Context, exoscaleDBaaS []*egoscale.DatabaseServi
...
+			if ds.salesOrder == "" {
+				itemGroup = fmt.Sprintf("APPUiO Cloud - Zone: %s / Namespace: %s", ds.clusterId, dbaasDetail.Namespace)
+				ds.salesOrder, err = controlAPI.GetSalesOrder(ctx, ds.controlApiClient, dbaasDetail.Organization)
+				if err != nil {
+					logger.Error(err, "Unable to sync DBaaS, cannot get salesOrder", "namespace", dbaasDetail.Namespace)
+					continue
+				}
+			}
+
+			// TODO zones and namespaces?
```
Comment (Kidswiss) [inline pkg/exoscale/dbaas.go:222]: Probably not necessary anymore?

**Review by Kidswiss — CHANGES_REQUESTED** [review]

File: pkg/exoscale/dbaas.go (hunk 5 above)
Comment (TheBigLee) [reply pkg/exoscale/dbaas.go:221]: Removed it

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**